### ISSUE TITLE
Shooter provider does not filter by language

### ISSUE THREAD

**Issue by MarioVilas** [Diaoul/subliminal#924]
When downloading subtitles using the Shooter provider, the language filter is ignored and instead any random language is chosen.

Can be reproduced with any video just as long as the Shooter provider is used.

```
$ subliminal download -l en -vvv My.Name.Is.Earl.S04E16.HDTV.XviD-LOL.avi
Collecting videos  [####################################]  100%                                          
1 video collected / 0 video ignored / 0 error
Downloading subtitles  [####################################]  100%                                          
Some providers have been discarded due to unexpected errors: legendastv
1 subtitle downloaded for My.Name.Is.Earl.S04E16.HDTV.XviD-LOL.avi
  - [ 99.7] English subtitle from shooter (match on hash)
$ head My.Name.Is.Earl.S04E16.HDTV.XviD-LOL.en.srt 
1
00:00:00,834 --> 00:00:03,378
<i>Z�chrann� t�m by cht�l zopakovat,</i>

2
00:00:03,462 --> 00:00:09,510
<i>�e sesuv p�dy je p��rodn� katastrofa,
ne obrovsk� jezero pln� alkoholu.</i>.

3
```

**Comment by ratoaq2** [comment]
Please provide the full logs. Instructions are here: https://github.com/Diaoul/subliminal/blob/master/CONTRIBUTING.md

**Comment by MarioVilas** [comment]
I got the following error trying to get the debug log with version 2.0.5:

`Error: no such option: --debug`

Strange because if I do a --help I can see the option listed there.

**Comment by MarioVilas** [comment]
Ok, I managed to get the debug log anyway by tweaking the source code a bit. Here it is.

[debug.txt](https://github.com/Diaoul/subliminal/files/2901153/debug.txt)

**Comment by ratoaq2** [comment]
shooter web site has news that it was shutdown around 2014 or 2015, but the API still works.
subliminal provider is sending the request with the language parameter. But I can't validate if that is what the api expects. In the end, it could also be that wrong language was registered in their website for that subtitle.

I'm not sure how to proceed here

**Comment by MarioVilas** [comment]
If the subtitles were uploaded with the wrong language, it must have happened to almost all of them, and with wildly varying languages. I suspect what's really happening is the API is either ignoring the language, or however the subtitles got uploaded they got English as a default and a lot of people didn't bother to change it.

Seems to me like this provider should be off by default.

Another choice would be to try to detect the language after downloading to make sure, using a library like langdetect (https://pypi.org/project/langdetect/) but this would necessarily be inaccurate. Could work as a warning to the user, though.

**Comment by ratoaq2** [comment]
I did some research and I believe it's better to have it disabled by default.

**Comment by sercankd** [comment]
I have the same issue, its downloading Spanish subtitles.

> D:\TorrentDownloads\The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEBRip.DDP5.1.x264-CasStudio[rarbg]>subliminal --debug download -l en "The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio.mkv"
> DEBUG:subliminal.cli:Collecting path The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio.mkv
> INFO:subliminal.core:Scanning video 'The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio.mkv' in ''
> DEBUG:subliminal.core:Size is 3279177385
> DEBUG:subliminal.core:Computed hashes {'opensubtitles': 'd146293782bdebae', 'shooter': 'a2810ee86238d684f22042d186babe87;48ca93bf3fe296f9202636460194c655;f34a3306ff4eedf9746a7265be280a52;8311a31486799d132d740829da9fc15d', 'thesubdb': '1e420deb3b3934470291d7bed155869b', 'napiprojekt': '9d5b9dba19d93c7dea2aa7dd84dfa4b1'}
> DEBUG:subliminal.core:Found subtitles {}
> INFO:subliminal.core:Refining video with metadata
> DEBUG:subliminal.refiners.metadata:Found resolution 1080p
> DEBUG:subliminal.refiners.metadata:Found video_codec h264
> DEBUG:subliminal.refiners.metadata:Embedded subtitle track name 'SDH' is not a valid language
> DEBUG:subliminal.refiners.metadata:Found embedded subtitle {<Language [und]>}
> INFO:subliminal.core:Refining video with tvdb
> INFO:subliminal.refiners.tvdb:Searching series 'The 100'
> DEBUG:subliminal.refiners.tvdb:Found 96 results
> DEBUG:subliminal.refiners.tvdb:Found exact match on series 'The 100'
> DEBUG:subliminal.refiners.tvdb:Discarding series name 'Aquaman' mismatch on year 2006
> DEBUG:subliminal.refiners.tvdb:Discarding series name 'Cartoons That Time Forgot: The UB Iwerks Collection, Vol. 1' mismatch on year 1935
> DEBUG:subliminal.refiners.tvdb:Discarding series name 'Hysteria' mismatch on year 2014
> DEBUG:subliminal.refiners.tvdb:Discarding series name 'Millennium' mismatch on year 2010
> DEBUG:subliminal.refiners.tvdb:Discarding series name 'Power' mismatch on year 2014
> DEBUG:subliminal.refiners.tvdb:Discarding series name 'Mob Psycho 100' mismatch on year 2018
> DEBUG:subliminal.refiners.tvdb:Found series {'id': 268592, 'seriesName': 'The 100', 'aliases': ['The Hundred', 'The One Hundred'], 'banner': 'graphical/268592-g12.jpg', 'seriesId': '183133', 'status': 'Continuing', 'firstAired': '2014-03-19', 'network': 'The CW', 'networkId': '', 'runtime': '45', 'genre': ['Action', 'Drama', 'Science-Fiction', 'Suspense', 'Thriller'], 'overview': "Set ninety-seven years after a nuclear war has destroyed civilization, when a spaceship housing humanity's lone survivors sends one hundred juvenile delinquents back to Earth, in hopes of possibly re-populating the planet.\r\n", 'lastUpdated': 1562935665, 'airsDayOfWeek': 'Tuesday', 'airsTime': '9:00 PM', 'rating': 'TV-14', 'imdbId': 'tt2661044', 'zap2itId': 'SH01740398', 'added': '2013-04-09 18:54:34', 'addedBy': 141901, 'siteRating': 8.3, 'siteRatingCount': 163, 'slug': 'the-100'}
> INFO:subliminal.refiners.tvdb:Getting series episode 6x9
> DEBUG:subliminal.refiners.tvdb:Found episode {'id': 7223605, 'airedSeason': 6, 'airedSeasonID': 774388, 'airedEpisodeNumber': 9, 'episodeName': 'What You Take With You', 'firstAired': '2019-07-09', 'guestStars': [], 'director': '||', 'directors': [], 'writers': [], 'overview': 'Bellamy must venture into enemy territory with an unlikely companion. Meanwhile, Octavia is forced to confront her past.', 'language': {'episodeName': 'en', 'overview': 'en'}, 'productionCode': '', 'showUrl': '', 'lastUpdated': 1562165659, 'dvdDiscid': '', 'dvdSeason': 0, 'dvdEpisodeNumber': 0, 'dvdChapter': None, 'absoluteNumber': 0, 'filename': 'episodes/268592/7223605.jpg', 'seriesId': 268592, 'lastUpdatedBy': 504472, 'airsAfterSeason': None, 'airsBeforeSeason': None, 'airsBeforeEpisode': None, 'thumbAuthor': 504472, 'thumbAdded': '2019-07-03 07:54:19', 'thumbWidth': '400', 'thumbHeight': '225', 'imdbId': '', 'siteRating': 0, 'siteRatingCount': 0}
> INFO:subliminal.core:Refining video with omdb
> DEBUG:subliminal.refiners.omdb:No need to search
> 1 video collected / 0 video ignored / 0 error
> INFO:subliminal.core:Listing subtitles with provider 'addic7ed' and languages {<Language [en]>}
> INFO:subliminal.core:Initializing provider addic7ed
> INFO:subliminal.core:Listing subtitles with provider 'legendastv' and languages {<Language [en]>}
> INFO:subliminal.core:Initializing provider legendastv
> INFO:subliminal.core:Listing subtitles with provider 'opensubtitles' and languages {<Language [en]>}
> INFO:subliminal.core:Listing subtitles with provider 'podnapisi' and languages {<Language [en]>}
> INFO:subliminal.core:Initializing provider opensubtitles
> DEBUG:subliminal.providers.addic7ed:Getting show id with year
> INFO:subliminal.core:Listing subtitles with provider 'shooter' and languages {<Language [en]>}
> INFO:subliminal.core:Initializing provider podnapisi
> INFO:subliminal.providers.opensubtitles:Logging in
> INFO:subliminal.core:Skipping provider 'subscenter': no language to search for
> DEBUG:subliminal.providers.addic7ed:Getting show id
> INFO:subliminal.core:Listing subtitles with provider 'thesubdb' and languages {<Language [en]>}
> INFO:subliminal.core:Initializing provider shooter
> INFO:subliminal.core:Listing subtitles with provider 'tvsubtitles' and languages {<Language [en]>}
> INFO:subliminal.providers.podnapisi:Searching subtitles {'sXML': 1, 'sL': 'en', 'sK': 'The 100', 'sTS': 6, 'sTE': 9, 'sY': 2014}
> WARNING:subliminal.providers.addic7ed:Series not found in show ids
> INFO:subliminal.core:Initializing provider thesubdb
> INFO:subliminal.core:Initializing provider tvsubtitles
> DEBUG:subliminal.providers.shooter:Searching subtitles {'filehash': 'a2810ee86238d684f22042d186babe87;48ca93bf3fe296f9202636460194c655;f34a3306ff4eedf9746a7265be280a52;8311a31486799d132d740829da9fc15d', 'pathinfo': 'D:\TorrentDownloads\The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEBRip.DDP5.1.x264-CasStudio[rarbg]\The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio.mkv', 'format': 'json', 'lang': 'eng'}
> INFO:subliminal.providers.thesubdb:Searching subtitles {'action': 'search', 'hash': '1e420deb3b3934470291d7bed155869b'}
> ERROR:subliminal.providers.addic7ed:No show id found for 'The 100' ({'year': 2014, 'country': None})
> INFO:subliminal.providers.tvsubtitles:Getting the page for episode 75017
> DEBUG:subliminal.providers.opensubtitles:Logged in with token '6-dJsjs946apftcwfv9aUGsGLf5'
> INFO:subliminal.providers.opensubtitles:Searching subtitles [{'moviehash': 'd146293782bdebae', 'moviebytesize': '3279177385', 'sublanguageid': 'eng'}, {'tag': 'The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio.mkv', 'sublanguageid': 'eng'}, {'query': 'The 100', 'season': 6, 'episode': 9, 'sublanguageid': 'eng'}]
> DEBUG:subliminal.providers.thesubdb:No subtitles found
> DEBUG:subliminal.providers.opensubtitles:Found subtitle <OpenSubtitlesSubtitle '1956399592' [en]> by fulltext
> DEBUG:subliminal.providers.opensubtitles:Found subtitle <OpenSubtitlesSubtitle '1956399590' [en]> by fulltext
> DEBUG:subliminal.providers.podnapisi:Found subtitle <PodnapisiSubtitle 'qCBG' [en]>
> DEBUG:subliminal.providers.podnapisi:Found subtitle <PodnapisiSubtitle 'pyBG' [en]>
> DEBUG:subliminal.providers.tvsubtitles:Found subtitle <TVsubtitlesSubtitle '343513' [en]>
> DEBUG:subliminal.providers.tvsubtitles:Found subtitle <TVsubtitlesSubtitle '343514' [en]>
> DEBUG:subliminal.providers.tvsubtitles:Found subtitle <TVsubtitlesSubtitle '343606' [es]>
> INFO:subliminal.score:Computing score of <OpenSubtitlesSubtitle '1956399592' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'episode', 'year', 'series', 'season', 'title'}
> DEBUG:subliminal.score:Adding title match equivalent
> DEBUG:subliminal.score:Matched hearing_impaired
> INFO:subliminal.score:Computed score 331 with final matches {'episode', 'year', 'hearing_impaired', 'series', 'season', 'title'}
> INFO:subliminal.score:Computing score of <OpenSubtitlesSubtitle '1956399590' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'episode', 'year', 'series', 'season', 'title'}
> DEBUG:subliminal.score:Adding title match equivalent
> INFO:subliminal.score:Computed score 330 with final matches {'episode', 'year', 'series', 'season', 'title'}
> INFO:subliminal.score:Computing score of <PodnapisiSubtitle 'qCBG' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'episode', 'year', 'series', 'season'}
> INFO:subliminal.score:Computed score 330 with final matches {'episode', 'year', 'series', 'season'}
> INFO:subliminal.score:Computing score of <PodnapisiSubtitle 'pyBG' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'episode', 'year', 'series', 'season'}
> DEBUG:subliminal.score:Matched hearing_impaired
> INFO:subliminal.score:Computed score 331 with final matches {'episode', 'year', 'hearing_impaired', 'series', 'season'}
> INFO:subliminal.score:Computing score of <ShooterSubtitle 'https://www.shooter.cn/api/subapi.php?fetch=MTU2MzA0MDkwMXxJZktDMXgtMUZpOXBhZ0UtVE5jUFNGZURWTDFieElnZWg0cmhVNjh3YjZWMG9mUS1pVVZwT2FveFJ4MHBFNWFJQTZkMi1FNzhEYlhSYkJrMlVGejE0VllBemFOLTlIY1Q4QlZCamxPakxpcG1IWmNfWHp1VDhMd081R29vR1RROUl0dTEzR1Fib2ZmbXlNUDdfWU5RdWZ4b2tkVlYxcUF1Z3BLNEZ4UktDY1ZKdFFFLVNTaXNDZjBHUXlxajdaUWlMWUNTMnJISmJtZDBBTFAwM1lXNmY1YnZfVzY5VzlnNnAxZndvaFNlck10RmM4ZzVRTmNPX2ZGMUVtdEFRaGFwZHlRQmg1T1RLaGpGSEVKcUF4ejFPaWV4Y0FCeDJLYWlzTGJsUHVXZVlzVWU0VzJRZlpWUFdFTXdyTWNxMngzSEN6WkNmWVdjelJlZWthdVRScTBFNWxGa0VXUWpUT0U9fDjzsH9Lv27D5XrKpAktvJtXKsaftUWJgOgZ3yGgKVDy&nonce=L%AB%8A%C2S%E3%5Bo%F3_%2A%9Bry%29%D4' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'hash'}
> DEBUG:subliminal.score:Keeping only hash match
> DEBUG:subliminal.score:Matched hearing_impaired
> INFO:subliminal.score:Computed score 360 with final matches {'hash', 'hearing_impaired'}
> INFO:subliminal.score:Computing score of <TVsubtitlesSubtitle '343513' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'episode', 'season', 'year', 'series'}
> DEBUG:subliminal.score:Matched hearing_impaired
> INFO:subliminal.score:Computed score 331 with final matches {'episode', 'year', 'hearing_impaired', 'series', 'season'}
> INFO:subliminal.score:Computing score of <TVsubtitlesSubtitle '343514' [en]> for video <Episode ['The 100', 2014, 6x9]> with {'hearing_impaired': False}
> DEBUG:subliminal.score:Using scores {'hash': 359, 'series': 180, 'year': 90, 'season': 30, 'episode': 30, 'release_group': 15, 'format': 7, 'audio_codec': 3, 'resolution': 2, 'video_codec': 2, 'hearing_impaired': 1}
> DEBUG:subliminal.score:Found matches {'episode', 'season', 'year', 'series'}
> DEBUG:subliminal.score:Matched hearing_impaired
> INFO:subliminal.score:Computed score 331 with final matches {'episode', 'year', 'hearing_impaired', 'series', 'season'}
> INFO:subliminal.core:Downloading subtitle <ShooterSubtitle 'https://www.shooter.cn/api/subapi.php?fetch=MTU2MzA0MDkwMXxJZktDMXgtMUZpOXBhZ0UtVE5jUFNGZURWTDFieElnZWg0cmhVNjh3YjZWMG9mUS1pVVZwT2FveFJ4MHBFNWFJQTZkMi1FNzhEYlhSYkJrMlVGejE0VllBemFOLTlIY1Q4QlZCamxPakxpcG1IWmNfWHp1VDhMd081R29vR1RROUl0dTEzR1Fib2ZmbXlNUDdfWU5RdWZ4b2tkVlYxcUF1Z3BLNEZ4UktDY1ZKdFFFLVNTaXNDZjBHUXlxajdaUWlMWUNTMnJISmJtZDBBTFAwM1lXNmY1YnZfVzY5VzlnNnAxZndvaFNlck10RmM4ZzVRTmNPX2ZGMUVtdEFRaGFwZHlRQmg1T1RLaGpGSEVKcUF4ejFPaWV4Y0FCeDJLYWlzTGJsUHVXZVlzVWU0VzJRZlpWUFdFTXdyTWNxMngzSEN6WkNmWVdjelJlZWthdVRScTBFNWxGa0VXUWpUT0U9fDjzsH9Lv27D5XrKpAktvJtXKsaftUWJgOgZ3yGgKVDy&nonce=L%AB%8A%C2S%E3%5Bo%F3_%2A%9Bry%29%D4' [en]>
> INFO:subliminal.providers.shooter:Downloading subtitle <ShooterSubtitle 'https://www.shooter.cn/api/subapi.php?fetch=MTU2MzA0MDkwMXxJZktDMXgtMUZpOXBhZ0UtVE5jUFNGZURWTDFieElnZWg0cmhVNjh3YjZWMG9mUS1pVVZwT2FveFJ4MHBFNWFJQTZkMi1FNzhEYlhSYkJrMlVGejE0VllBemFOLTlIY1Q4QlZCamxPakxpcG1IWmNfWHp1VDhMd081R29vR1RROUl0dTEzR1Fib2ZmbXlNUDdfWU5RdWZ4b2tkVlYxcUF1Z3BLNEZ4UktDY1ZKdFFFLVNTaXNDZjBHUXlxajdaUWlMWUNTMnJISmJtZDBBTFAwM1lXNmY1YnZfVzY5VzlnNnAxZndvaFNlck10RmM4ZzVRTmNPX2ZGMUVtdEFRaGFwZHlRQmg1T1RLaGpGSEVKcUF4ejFPaWV4Y0FCeDJLYWlzTGJsUHVXZVlzVWU0VzJRZlpWUFdFTXdyTWNxMngzSEN6WkNmWVdjelJlZWthdVRScTBFNWxGa0VXUWpUT0U9fDjzsH9Lv27D5XrKpAktvJtXKsaftUWJgOgZ3yGgKVDy&nonce=L%AB%8A%C2S%E3%5Bo%F3_%2A%9Bry%29%D4' [en]>
> INFO:subliminal.subtitle:Guessing encoding for language en
> DEBUG:subliminal.subtitle:Trying encodings ['utf-8', 'latin-1']
> INFO:subliminal.subtitle:Guessed encoding latin-1
> INFO:subliminal.subtitle:Guessing encoding for language en
> DEBUG:subliminal.subtitle:Trying encodings ['utf-8', 'latin-1']
> INFO:subliminal.subtitle:Guessed encoding latin-1
> DEBUG:subliminal.core:All languages downloaded
> Downloading subtitles [####################################] 100% The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEDownloading subtitles [####################################] 100%
> DEBUG:subliminal.core:Terminating initialized providers
> INFO:subliminal.core:Terminating provider addic7ed
> INFO:subliminal.core:Terminating provider legendastv
> INFO:subliminal.core:Terminating provider podnapisi
> INFO:subliminal.core:Terminating provider shooter
> INFO:subliminal.core:Terminating provider thesubdb
> INFO:subliminal.core:Terminating provider tvsubtitles
> INFO:subliminal.core:Terminating provider opensubtitles
> INFO:subliminal.providers.opensubtitles:Logging out
> DEBUG:subliminal.providers.opensubtitles:Logged out
> INFO:subliminal.core:Saving <ShooterSubtitle 'https://www.shooter.cn/api/subapi.php?fetch=MTU2MzA0MDkwMXxJZktDMXgtMUZpOXBhZ0UtVE5jUFNGZURWTDFieElnZWg0cmhVNjh3YjZWMG9mUS1pVVZwT2FveFJ4MHBFNWFJQTZkMi1FNzhEYlhSYkJrMlVGejE0VllBemFOLTlIY1Q4QlZCamxPakxpcG1IWmNfWHp1VDhMd081R29vR1RROUl0dTEzR1Fib2ZmbXlNUDdfWU5RdWZ4b2tkVlYxcUF1Z3BLNEZ4UktDY1ZKdFFFLVNTaXNDZjBHUXlxajdaUWlMWUNTMnJISmJtZDBBTFAwM1lXNmY1YnZfVzY5VzlnNnAxZndvaFNlck10RmM4ZzVRTmNPX2ZGMUVtdEFRaGFwZHlRQmg1T1RLaGpGSEVKcUF4ejFPaWV4Y0FCeDJLYWlzTGJsUHVXZVlzVWU0VzJRZlpWUFdFTXdyTWNxMngzSEN6WkNmWVdjelJlZWthdVRScTBFNWxGa0VXUWpUT0U9fDjzsH9Lv27D5XrKpAktvJtXKsaftUWJgOgZ3yGgKVDy&nonce=L%AB%8A%C2S%E3%5Bo%F3_%2A%9Bry%29%D4' [en]> to 'The.100.S06E09.What.You.Take.With.You.1080p.AMZN.WEB-DL.DD+5.1.H.264-CasStudio.en.srt'
> Downloaded 1 subtitle

**Comment by ratoaq2** [comment]
Fixed by f0cc387c65025d1b0cd64b7522b80792ded9a1c3
